pkgs/desktops/gnome-2/platform/libgnomeprint/default.nix

```diff
@@ -9,10 +9,16 @@ stdenv.mkDerivation rec {
     sha256 = "1034ec8651051f84d2424e7a1da61c530422cc20ce5b2d9e107e1e46778d9691";
   };
 
-  patches = [ ./bug653388.patch ];
+  patches = [
+    ./bug653388.patch
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/pld-linux/libgnomeprint/54c0f9c3675b86c53f6d77a5bc526ce9ef0e38cd/bison3.patch";
```

Comment:
Could we use the short URL, instead of the redirect?

Reply:
I don't see any redirects, maybe it would be nicer to include patch in nixpkgs source?

Reply:
It is redirected to githubusercontent.com from the URL I suggested previously https://github.com/NixOS/nixpkgs/pull/77051#discussion_r363266798

`fetchpatch` is preferred over including the patch into repo, as long as the URL is stable and the patch applies because patches bloat the repo and are often left behind when not needed anymore.

Reply:
Sorry but I still don't see any redirects, am I missing something ?
```
$ curl -i https://raw.githubusercontent.com/pld-linux/libgnomeprint/54c0f9c3675b86c53f6d77a5bc526ce9ef0e38cd/bison3.patch
HTTP/1.1 200 OK
Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
Strict-Transport-Security: max-age=31536000
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-XSS-Protection: 1; mode=block
ETag: "1cd24aa0b36cef0e4f1f75d2fc67a7c39e237c03eccec949cde4a43031cef9b5"
Content-Type: text/plain; charset=utf-8
Cache-Control: max-age=300
X-Geo-Block-List:
X-GitHub-Request-Id: AAA4:4EFD:6B46:8A15:5E14F4CD
Content-Length: 773
Accept-Ranges: bytes
Date: Tue, 07 Jan 2020 21:14:54 GMT
Via: 1.1 varnish
Connection: keep-alive
X-Served-By: cache-fra19173-FRA
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1578431694.169775,VS0,VE175
Vary: Authorization,Accept-Encoding
Access-Control-Allow-Origin: *
X-Fastly-Request-ID: be2fb08d025b3fa84c6f1aad90fbed1e36239a6f
Expires: Tue, 07 Jan 2020 21:19:54 GMT
Source-Age: 0

--- libgnomeprint-2.18.8/libgnomeprint/grammar.y.orig   2010-02-09 13:32:51.000000000 +0100
+++ libgnomeprint-2.18.8/libgnomeprint/grammar.y        2013-11-03 21:44:59.859877696 +0100
@@ -6,8 +6,6 @@
 #include <libgnomeprint/types.h>
 #include <libgnomeprint/gnome-print-filter.h>

-#define YYPARSE_PARAM graph
-
 static void
 set_value_from_string (GParamSpec *pspec, GValue *v, const gchar *s)
 {
@@ -99,7 +97,7 @@
 }

 static int yylex (void *lvalp);
-static int yyerror (const char *s);
+static int yyerror (graph_t *g, const char *s);
 %}

 %union {
@@ -117,6 +115,7 @@
 %type <p> pool

 %pure_parser
+%parse-param { graph_t *graph }

 %start graph
 %%
@@ -185,7 +184,7 @@
 %%

 static int
-yyerror (const char *s)
+yyerror (graph_t *g, const char *s)
 {
        return -1;
 }
```

Reply:
In the comment, I suggested a different URL that redirects to this one.

Reply:
hmm, I really need vacation :) thank you for your patience